scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -27,89 +29,88 @@
     public FileSystemService() {
     }
 
-    /**
-     * Creates the dataset (and its required directories) on the filesystem.
-     *
-     * @param id       The id of the dataset.
-     * @param username The username of the dataset.
-     * @throws DatasetStorageException The exception thrown if the directory could not be created.
-     */
+    public String getRootLocationString() {
+        if (rootLocationString != null) {
+            return rootLocationString;
+        } else {
+            return "";
+        }
+    }
+
     @Override
-    public void addDatasetDirectories(String id, String username) throws DatasetStorageException {
-        // First, the correct directory must be created
-        Path path = createPath(username, id);
+    public boolean addDirectories(String username, String type, String id, String version) throws
+            FileSystemStorageException {
+        if (username == null) {
+            return false;
+        }
+        if ((type == null || id == null || version == null) &&
+                (type != null || id != null || version != null)) {
+            return false;
```

Comment:
If I'm not mistaken `createPath` will just return null if these conditions are not met? Why not just check for null after line 50?